docs/dev-notes/GIT-WORKFLOW.md

```diff
@@ -46,114 +41,164 @@ With all of that explained, lets go over some common scenarios that you will enc
+
+   Know that eventually, all of your branches that you make for `edit-profile` will eventually get merged into the branch `your-name/edit-profile`. 
+
+4. Let's say that 2 weeks have gone by. You have completed all of the work for editing a profile. This big feature is done. Congrats! 
+   Make sure that all of the pull requests that you have made up to this point are merged into the branch `your-name/edit-profile`. This means that `your-name/edit-profile` is complete and contains all of the code for this big feature. Next is to merge it into `develop`! 
+   
+   Create a new pull request for `your-name/edit-profile` setting the base branch to `develop`. No need to get any pull request reviews for this pull request because all of the commits you made up to this point have been reviewed. So, you can go ahead and just merge the pull request into `develop` after all of the GitHub Status Checks in the pull request are successful. 
+   
+   ![CleanShot 2022-06-29 at 11.12.07@2x](../img/github_pr_failed_status_checks.png)
+
+> Here, you see that this pull request on GitHub has failed status checks. Make sure to fix all of the errors in tests and lint before merging your pull request. 
+
+![CleanShot 2022-06-29 at 11.13.16@2x](../img/github_pr_successful_status_checks.png)
+
+> This pull request on GitHub has all of the status checks passing. That means the pull request is ready to merge! 
+
+
 
 ## Fix a bug 
 
-Where did you find this bug? 
-* **In an `alpha` or `beta` release** of the software (the version of the release ends with `-alpha.X` or `-beta.X` such as `1.0.0-alpha.1`)? 
-If so, make a new git branch off of the `alpha` or `beta` branch, fix the bug, then make a pull request merging your pull request into the `alpha` or `beta` branch. 
+Where does this bug exist? Depending on where the bug exists will determine what you should do.
+
+> Note: When making a bug fix pull request, it's preferred that you include automated tests (unit test function, integration test, etc) in this pull request that reproduces the bug. This is to help us feel confident the bug is indeed fixed and it will not come up again in the future. 
+
+
+
+## Where did you find this bug? 
+
+### In an `alpha` or `beta` release of the software (the version of the release ends with `-alpha.X` or `-beta.X` such as `1.0.0-alpha.1`)
```

Comment:
IMO it is good to mention here that these versions are taken care by CI/CD bots, just as a side note.